### PR TITLE
segment should remove all trailing/leading slashes

### LINKED ISSUE #236: Should segment() take care of leading slash?
I'm using URI.js to append a cdn host to URLs, the problem is that some assets have a leading slash and other don't.

Should `segment()` method take care of this?

## Example

```
URI('https://google.com').segment('font.ttf').toString() //-> https://google.com/font.ttf
URI('https://google.com').segment('/font.ttf').toString() //-> https://google.com//font.ttf
```
Result of second example should be like the first one.

Thanks!

### PR DESCRIPTION
**Pull request by orlando**
# Description

This PR removes leading and trailing slashes from `segment`

Fixes #236

### CODE REVIEW

File: src/URI.js [hunk 1]
```diff
@@ -1553,6 +1553,7 @@
     var path = this.path();
     var absolute = path.substring(0, 1) === '/';
     var segments = path.split(separator);
+    var noLeadTrailSlashRegex = /^\/*|\/*$/g;
```
Comment (rodneyrehm) [inline src/URI.js]: shouldn't that be `/^\/+|\/+$/g;` (`+` (one or more) instead of `*` (zero or more))?

File: src/URI.js [hunk 2]
```diff
@@ -1615,6 +1616,13 @@
       segments.unshift('');
     }
 
+    // remove leading and trailing slashes
+    // from segment
+    for (var i=0; i < segments.length; i++) {
```
Comment (rodneyrehm) [inline src/URI.js]: isn't this overkill, as you only need to replace the `v` upon setting a value?

File: src/URI.js (hunk 1 above)
Comment (orlando) [inline src/URI.js]: yes you are right, I'll fix this

File: src/URI.js (hunk 2 above)
Comment (orlando) [inline src/URI.js]: the problem is that we are setting this value in multiple places in the `if` statement in this method, and I didn't want to add more complexity to that part of the code.

I can change the code to do this replace in each branch if you think that's better.

Comment (rodneyrehm) [inline src/URI.js]: the lines we're writing in are 1596, 1602, 1604, 1609. For lines 1602 and 1604 a single sanitization can be done on line 1607+. That makes three times `trimSlashes(v)` and one declaration of the function somewhere at the top (likely line 177, as the last internal utility function).

```js
function trimSlashes(text) {
  return text.replace(/^\/+|\/+$/g, '');
}
```
doesn't sound like making that if-from-hell any more complex… (looking at that code, a refactoring might be in order anyway…)

Comment (orlando) [inline src/URI.js]: :+1: got it, cool I'll make the changes

**Comment by rodneyrehm** [comment]
ping @orlando :)

**Comment by orlando** [comment]
@rodneyrehm PR updated

**Comment by rodneyrehm** [comment]
thank you very much! :)

**Comment by rodneyrehm** [comment]
released in [v1.17.0](https://github.com/medialize/URI.js/releases/tag/v1.17.0)

**Comment by orlando** [comment]
:fire:

**Pull request merged**